app/delivery/send_to_providers.py

```diff
@@ -65,8 +66,24 @@ def send_sms_to_provider(notification):
                 # providers as a slow down of our providers can cause us to run out of DB connections
                 # Therefore we pull all the data from our DB models into `send_sms_kwargs`now before
                 # closing the session (as otherwise it would be reopened immediately)
+
+                # We start by trying to get the phone number from a job in s3.  If we fail, we assume
+                # the phone number is for the verification code on login, which is not a job.
+                my_phone = None
+                try:
+                    my_phone = get_phone_number_from_s3(
+                        notification.service_id,
+                        notification.job_id,
+                        notification.job_row_number,
+                    )
+                except BaseException:
+                    my_phone = redis_store.get(f"2facode_{notification.id}")
+                    if my_phone:
+                        my_phone = my_phone.decode("utf-8")
+                if my_phone is None:
+                    raise Exception("what happened to the phone number")
```

Comment:
I think it's appropriate to raise the exception here for this, though perhaps we ought to be clearer on what's happening (especially given our conversation earlier today on log messages! 😆)?

```suggestion
                    raise Exception(f"The phone number for (Service ID: {notification.service_id}; Job ID: {notification.job_id}; Job Row Number {notification.job_row_number} was not found.")
```

(I'm not sure if this would be exposing too much info - I don't think so with the Service and Job IDs at least.)

That said, does this mean we should also get some additional error handling in place elsewhere to take care of this exception if it's ever thrown?

Reply:
I think your suggestion is good.  I don't think it exposes too much and besides it "should never happen", etc.